quill-core/src/main/scala/io/getquill/norm/NormalizeReturning.scala

```diff
@@ -19,9 +19,15 @@ object NormalizeReturning {
 
   private def filterReturnedColumn(assignment: Assignment, column: Ast): Option[Assignment] =
     (assignment, column) match {
-      case (Assignment(_, Property(_, p1), _), Property(_, p2)) if (p1 == p2) =>
+      case (Assignment(_, p1: Property, _), p2: Property) if p1.name == p2.name && isSameProperties(p1, p2) =>
         None
       case (assignment, column) =>
         Some(assignment)
     }
+
+  private def isSameProperties(p1: Property, p2: Property): Boolean = (p1.ast, p2.ast) match {
+    case (_: Ident, _: Ident)           => p1.name == p2.name
```

Comment:
idents are always different since they comes from different scopes hence we ignore them